src/wms/src/main/java/org/geoserver/wms/legendgraphic/BufferedImageLegendGraphicBuilder.java

```diff
@@ -671,7 +675,7 @@ private RenderedImage getLayerLegend(LegendRequest legend, int w, int h, boolean
      */
     private BufferedImage mergeGroups(List<RenderedImage> imageStack, Rule[] rules, GetLegendGraphicRequest req,
             boolean forceLabelsOn, boolean forceLabelsOff) {
-        LegendMerger.MergeOptions options = LegendMerger.MergeOptions.createFromRequest(imageStack, 0, 0, 0, req, forceLabelsOn, forceLabelsOff);
+        LegendMerger.MergeOptions options = LegendMerger.MergeOptions.createFromRequest(imageStack, 0, 0, 0, 0, req, forceLabelsOn, forceLabelsOff);
```

Comment:
If I understood correctly the idea is that the label margin default value is 3, right ? But in this case the label margin value will have no effect so we set it to zero, right ?